[Zheng Wentao] iP

Zhengwinter opened the pull request.

Zhengwinter committed:
Level 0: Renamed, added greet, exit

Zhengwinter committed:
Implement echo, restructure hi and bye

Zhengwinter committed:
Implement add task

Used String array to implement the task list, used continue statement to execute the list command

Zhengwinter committed:
Restructure Task as a class, implement task marking

Zhengwinter committed:
Revert "Restructure Task as a class, implement task marking"

This reverts commit cb2d17f9cfc7b755368992e6a0a258d4ba16044f.

Zhengwinter committed:
Explain methods and Task class

Made some changes according to Java coding standard

Zhengwinter committed:
Merge branch 'master' of https://github.com/Zhengwinter/ip

# Conflicts:
#	src/main/java/Winter.java

Zhengwinter committed:
Correct code version

Zhengwinter committed:
Refactor code

Restructure code using concepts of abstraction. Applied enumeration to handle invalid inputs from the user

Zhengwinter committed:
Implement ToDo, Deadline and Event classes

Used inheritance to from the Task superclass to implement the 3 types of classes. Further restructured code and attempted to minimize repetition

runxinghuan commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,3 @@
+public enum Commands {
+    bye,,list,mark,unmark
```
I think there is a redundant comma here.

runxinghuan commented on src/main/java/Winter.java:
```diff
@@ -0,0 +1,161 @@
+    }
+    // Class for Task object
+    private static class Task {
+        boolean marked;
```
This boolean should be named to sound like booleans. Consider changing it to "isMarked"?

runxinghuan commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,2 @@
+package PACKAGE_NAME;public class Deadline {
```
I like how you named your classes, which are all in PascalCase.

runxinghuan commented on src/main/java/Winter.java:
```diff
@@ -0,0 +1,161 @@
+
+    }
+    // Method for adding tasks by handling user commands
+    private static void addTasks() {
```
I think plural form is used on names representing a collection of objects only. Maybe you should use singular form here?

runxinghuan commented on src/main/java/Winter.java:
```diff
@@ -0,0 +1,161 @@
+            case "LIST":
+                displayList(taskList,taskIndex);
+                continue;
+
```
I notice that you did not add a `default` case in the `switch` statement. Did you do this intentionally?

runxinghuan reviewed (commented):
Looks good to me overall, your code was easy to read. I like that you named all your classes in PascalCase and methods in camelCase. Good job!

lordgareth10 commented on src/main/java/Winter.java:
```diff
@@ -0,0 +1,161 @@
+        System.out.print(line);
+        System.out.println(farewell);
+        System.out.print(line);
+    }
```
I like how the naming conventions are followed well! The method names use verbs for actions like sayHi and sayBye which is compliant with the coding standards.

lordgareth10 commented on src/main/java/Winter.java:
```diff
@@ -0,0 +1,161 @@
+            System.out.println(echoLine);
+            System.out.print(line);
+            echoLine = input.nextLine();
+        }
```
The use of while(true) could pose some readability issues as it is not clear when the loop will terminate. You could consider refactoring the code to check for the if condition inside the conditional part of the while loop.

lordgareth10 commented on src/main/java/Winter.java:
```diff
@@ -0,0 +1,161 @@
+            taskList[taskIndex] = newTask;
+
+            switch(command) {
+            // Cases include farewell and list commands
```
I like how there is documentation to tell the reader what the case break block is doing here!

lordgareth10 commented on src/main/java/Winter.java:
```diff
@@ -0,0 +1,161 @@
+                continue;
+            case "unmark":
+                taskList[Integer.parseInt(command.substring(7,command.length()))-1].unmark();
+                continue;
```
You could consider using an extra variable to assign the parsed input to as this will make it more obvious what exactly the method mark() is being used on. For example, 
parsedIndex = Integer.parseInt(command.substring(5,command.length()))-1;
taskList[parsedIndex].mark();

lordgareth10 reviewed (commented):
Overall, I found the code easy to read and understand due to the addition of comments in various places. Classes were named in PascalCase, and variables/methods in camelCase. However there were some readability issues which I have pointed out and gave suggestions on how to fix.

Zhengwinter committed:
Add automated Text UI testing

Include both .bat and .sh file for Window/Mac or Linux users respectively.

Zhengwinter committed:
Implement automated testing

Modify shellscript files for testing and added test cases

Zhengwinter committed:
Handle exceptions, refactor code

Implemented try catch and throw for checked exceptions regarding the inputs. Refactored code according to peer reviews.

Zhengwinter committed:
Handle exceptions, refactor code

Implemented try catch and throw for checked exceptions regarding the inputs. Refactored code according to peer reviews.

Zhengwinter committed:
Syntax fixes

Minor syntax fixes

Zhengwinter committed:
Merge branch 'master' into branch-Level-5

"Add exception handling, and bug fixes"

Zhengwinter committed:
Merge branch

Zhengwinter committed:
Divide classes into packages

Divided classes into tasks, exceptions packages and put those together with the Winter, Manager and Command enum into a package named winter

Zhengwinter committed:
Fix test script path

Zhengwinter committed:
Fix minor output display issue

Zhengwinter committed:
Refactor code, fix bugs

Refactored code to use overloading for the toString method, fixed a erroneouos boolean flag

Zhengwinter committed:
Merge branch-Level-5 to master

Jonoans commented on src/main/java/winter/Manager.java:
```diff
@@ -0,0 +1,302 @@
+    private static int taskIndex = 0;
+    private static final Task[] taskList = new Task[100];
+
+    public static void acceptInput () {
```
Try to avoid long methods like this

Jonoans commented on src/main/java/winter/Manager.java:
```diff
@@ -0,0 +1,302 @@
+        Scanner input = new Scanner(System.in);
+        String inputString;
+        Commands action;
+        boolean flag = true;
```
Maybe `isFlag` would better conform to our coding standards

Jonoans commented on src/main/java/winter/Manager.java:
```diff
@@ -0,0 +1,302 @@
+            if (taskList[i].getType().equals("D")) {
+                System.out.println(taskList[i].getOrder()+1 + ". [D] " + taskList[i].doneCheckbox + " "
+                        + taskList[i].getTaskName() + " (by: " + taskList[i].getEndTime() +")");
+            }else if (taskList[i].getType().equals("E")){
+                System.out.println(taskList[i].getOrder()+1 + ". [E] " + taskList[i].doneCheckbox + " "
+                        + taskList[i].getTaskName() + " (from: " + taskList[i].getStartTime()
+                        + " to: " + taskList[i].getEndTime() + ")");
+            }else {
```
Missing spaces here

Jonoans commented on src/main/java/winter/task/Event.java:
```diff
@@ -0,0 +1,32 @@
+    protected String startTime;
+    protected String endTime;
+
+    public Event (int order,boolean isMarked, String eventName, String startTime, String endTime) {
```
Do pay mind to the spaces

Jonoans commented on src/main/java/winter/Manager.java:
```diff
@@ -0,0 +1,302 @@
+                displayList(taskList, taskIndex);
+                continue;
+            case MARK:
+                taskList[Integer.parseInt(inputString.substring(5)) - 1].mark();
```
Avoid magic numbers like this as it make not be immediately apparent what `5` means

Jonoans commented on src/main/java/winter/task/Task.java:
```diff
@@ -0,0 +1,66 @@
+    }
+    // Method for unmarking a class
+    public void unmark() {
+        String line = "-----------------------------------\n";
+        String indent = "   ";
```
Maybe these strings could be declared at the class-level as constants?

Jonoans commented on src/main/java/winter/task/Task.java:
```diff
@@ -0,0 +1,66 @@
+    int order;
+    String taskName;
+
+    public String doneCheckbox ="[ ]";
```
This seems to be repeated in line 31 as well, maybe consider using a constant variable to store `"[ ]"` and have doneCheckbox reference that constant for its initial value?

Jonoans commented on src/main/java/winter/Manager.java:
```diff
@@ -0,0 +1,302 @@
+    private static Commands classifyCommand(String inputString) {
+        switch (inputString) {
+        // Cases include farewell and list commands
+        case "bye":
+        case "Bye":
+        case "BYE":
+            return BYE;
+        case "list":
+        case "List":
+        case "LIST":
```
Understand what you are trying to do here, though maybe using `"some_string".toLowerCase()` can help you normalise the command word so you would not have to account for that many edge cases? This is outside of code quality / coding standard, just a tip that is hopefully helpful

Jonoans reviewed (commented):
Nice job with the functionality, however, do try to break long methods in your program into more digestible units. Try to avoid magic strings and magic numbers as well.

Zhengwinter committed:
Implement delete and restructure code

Used collections for taskList, implemented delete function

Zhengwinter committed:
Implement save function

Zhengwinter committed:
Fix merge conflicts and integrate the branches

Zhengwinter committed:
Remove error message

Zhengwinter committed:
Update tag A-Collections

Zhengwinter committed:
Refactor code, extract classes

Broke down the command into child classes, restructured code by applying OOP concepts

Zhengwinter committed:
Implement LocalDateTime object for Deadline time

Allow the chatbot to understand date and time that is entered as a string

Zhengwinter committed:
Implement find command

Added the find command and adjusted the code to integrate the command

Zhengwinter committed:
Write JavaDocs explaining methods and classes

Wrote JavaDocs comments for major classes and significant/important methods

Zhengwinter committed:
Merge pull request #1 from Zhengwinter/branch-Level-8

Implement LocalDateTime object for Deadline time

Zhengwinter committed:
Merge master to branch-Level-9

Zhengwinter committed:
Merge master with A-JavaDoc after LocalDateTime update

Zhengwinter committed:
Merge pull request #2 from Zhengwinter/branch-Level-9

Implement find command

Zhengwinter committed:
Merge master to branch-A-JavaDoc after updating with Find command

Zhengwinter committed:
Merge pull request #3 from Zhengwinter/branch-A-JavaDoc

Write JavaDocs explaining methods and classes

Zhengwinter committed:
Clear out data file

Zhengwinter committed:
Add user guide

Zhengwinter committed:
Fix duplicate display of text

Remove duplicate instantiation of UI class

Zhengwinter committed:
Fix bugs relating to the list

Zhengwinter committed:
Update readme
